Adds step-by-step LiveStore tutorial

nikolasburk opened the pull request:
I've created a step-by-step LiveStore tutorial.

A few open questions:

- I put the repo with the starter project (and final version of the app) into my own GitHub account: https://github.com/nikolasburk/livestore-tutorial-starter; should we move it into the LiveStore org?
- I put all the images into a dedicated `assets/tutorial` directory and linked them from there; some GIFs are a bit bigger (~5-6MB), is that a problem?
- I didn't use the type-checked `Snippet` components that are e.g. used in the "getting started" section. I didn't dive deeper but it seemed like they would only work if the code could be imported from the same repo, which isn't the case here.

Once this is merged, I'd also love to discuss how we can ensure the entire tutorial is being tested via CI if possible (e.g. via an agent).

nikolasburk committed:
welcome and prerequisites

nikolasburk committed:
first chapter

nikolasburk committed:
start second chapter

nikolasburk committed:
end third chapter

nikolasburk committed:
add new page for 4th chapter

nikolasburk committed:
end sixth and seventh chapter

nikolasburk committed:
refine and polish

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/livestorejs/livestore?pullRequest=798) <br/>All committers have signed the CLA.

nikolasburk committed:
Merge branch 'dev' into docs/tutorial

nikolasburk committed:
polish writing and remove unused files

nikolasburk committed:
Merge branch 'docs/tutorial' of github.com:nikolasburk/livestore into docs/tutorial

nikolasburk committed:
Merge branch 'dev' into docs/tutorial

nikolasburk commented on docs/src/content/docs/tutorial/1-setup-starter-project.mdx:
````diff
@@ -0,0 +1,93 @@
+
+```bash
+git clone git@github.com:nikolasburk/livestore-tutorial-starter.git livestore-todo-app
+```
````
````suggestion
Go ahead and clone its `starter` branch via the following command:

```bash
git clone \
  --branch starter \
  --single-branch git@github.com:nikolasburk/livestore-tutorial-starter.git \
  livestore-todo-app 
```
````

nikolasburk committed:
Apply suggestion from @nikolasburk

schickling committed:
Merge branch 'dev' into docs/tutorial

nikolasburk committed:
add starter project to examples and update setup instructions in tutorial

nikolasburk commented:
@schickling I just:
- added the `tutorial-starter` to the `examples` directory: https://github.com/nikolasburk/livestore/tree/docs/tutorial/examples/tutorial-starter
- updated the setup instructions to use:
  ```
  pnpm dlx @livestore/cli@dev create \
    --example tutorial-starter livestore-todo-app
  ```

nikolasburk committed:
add bun commands to tutorial

nikolasburk committed:
resolve merge conflict

nikolasburk committed:
add line breaks for code blocks with many dependencies

nikolasburk committed:
Merge branch 'dev' of github.com:nikolasburk/livestore into docs/tutorial

nikolasburk committed:
add tldraw diagrams

nikolasburk committed:
add tldraw diagrams and emebed via MDX

nikolasburk committed:
add raw tldraw files

nikolasburk commented on package.json:
```diff
@@ -9,6 +9,7 @@
     "@vitest/ui": "catalog:",
     "husky": "^9.1.7",
     "madge": "^8.0.0",
+    "puppeteer": "^24.27.0",
```
```suggestion
```

nikolasburk committed:
Apply suggestion from @nikolasburk

nikolasburk committed:
use biome instead of eslint in tutorial-starter

nikolasburk committed:
Merge branch 'docs/tutorial' of github.com:nikolasburk/livestore into docs/tutorial

nikolasburk committed:
fix lock file

nikolasburk committed:
Merge branch 'dev' into docs/tutorial

nikolasburk committed:
sync updates from dev branch

nikolasburk committed:
Merge branch 'docs/tutorial' of github.com:nikolasburk/livestore into docs/tutorial

nikolasburk committed:
Merge branch 'dev' into docs/tutorial
